test/src/responder/component.test.ts

```diff
@@ -0,0 +1,304 @@
+        const nextAnchorState = reducer.reduce(prevAnchorState, blocks[0]);
+
+        expect(nextAnchorState.identifier).to.equal(txID1.identifier);
+        expect(nextAnchorState.appointmentId).to.equal(appointmentId1);
+        expect(nextAnchorState.kind).to.equal(ResponderStateKind.Pending);
+    });
+
+    it("reduce transitions from pending to mined", () => {
+        const reducer = new ResponderAppointmentReducer(blockCache, txID1.identifier, appointmentId1, from1);
+
+        const prevAnchorState = reducer.getInitialState(blocks[0]);
+        const nextAnchorState = reducer.reduce(prevAnchorState, blocks[1]);
+
+        expect(nextAnchorState.kind).to.equal(ResponderStateKind.Mined);
+        if (nextAnchorState.kind === ResponderStateKind.Mined) {
+            expect(nextAnchorState.identifier).to.equal(txID1.identifier);
+            expect(nextAnchorState.appointmentId).to.equal(appointmentId1);
+            expect(nextAnchorState.blockMined).to.equal(txID1.tx.blockNumber);
+            expect(nextAnchorState.nonce).to.equal(txID1.tx.nonce);
+        }
+    });
+
+    it("reduce keeps mined as mined", () => {
+        const reducer = new ResponderAppointmentReducer(blockCache, txID1.identifier, appointmentId1, from1);
+
+        const prevAnchorState = reducer.getInitialState(blocks[0]);
+        const nextAnchorState = reducer.reduce(prevAnchorState, blocks[1]);
+        const nextNextAnchorState = reducer.reduce(nextAnchorState, blocks[2]);
+
+        expect(nextAnchorState).to.equal(nextNextAnchorState);
+    });
+
+    it("reduce doesnt mine tx from different address", () => {
+        const reducer = new ResponderAppointmentReducer(blockCache, txID1.identifier, appointmentId1, from1);
+
+        // setup pending
+        const prevAnchorState = reducer.getInitialState(blocks[0]);
+
+        // mine a block with the same txidentifier but a different 'from'
+        const nextAnchorState = reducer.reduce(prevAnchorState, blocks[2]);
+
+        expect(nextAnchorState.identifier).to.equal(txID1.identifier);
+        expect(nextAnchorState.appointmentId).to.equal(appointmentId1);
+        expect(nextAnchorState.kind).to.equal(ResponderStateKind.Pending);
+    });
+});
+
+const pendingAppointmentState = (appointmentId: string, data: string): PendingResponseState => {
+    const identifier = new PisaTransactionIdentifier(1, data, "to", new BigNumber(0), new BigNumber(200));
+    return {
+        appointmentId,
+        identifier,
+        kind: ResponderStateKind.Pending
+    };
+};
+
+const minedAppointmentState = (
+    appointmentId: string,
+    data: string,
+    blockMined: number,
+    nonce: number
+): MinedResponseState => {
+    const identifier = new PisaTransactionIdentifier(1, data, "to", new BigNumber(0), new BigNumber(200));
+    return {
+        appointmentId,
+        blockMined,
+        identifier,
+        kind: ResponderStateKind.Mined,
+        nonce
+    };
+};
```

Comment:
Not important, but I prefer to call these type of functions `makeX()` instead of `X()`.